### PR TITLE
Public Cloud: Remove publiccloud::ssh_interactive_init

### PR DESCRIPTION
**Pull request by pdostal**
The publiccloud::ssh_interactive_init module is useless and the name is just misleading.

- Related ticket: [poo#95780](https://progress.opensuse.org/issues/95780)
- Verification run: [Passing](https://pdostal-server.suse.cz/tests/235), [Failing in tunnel mode](https://pdostal-server.suse.cz/tests/234), [Failing out of tunnel mode](https://pdostal-server.suse.cz/tests/231)

## Notes
 * Fata console tests in Public Cloud must have `run_args`
 * We use `no_rollback => 1` but we were missing `fatal => 0`
   * See [os-autoinst/os-autoinst/autotest.pm#L384](https://github.com/os-autoinst/os-autoinst/blob/master/autotest.pm#L384)

### CODE REVIEW

**Review by asmorodskyi — APPROVED** [review]
LGTM

**Review by grisu48 — APPROVED** [review]
LGTM

File: lib/publiccloud/basetest.pm [hunk 1]
```diff
@@ -110,15 +110,19 @@ sub _cleanup {
     my ($self) = @_;
     die("Cleanup called twice!") if ($self->{cleanup_called});
     $self->{cleanup_called} = 1;
+    record_info 'XXX';
```
Comment (asmorodskyi) [inline lib/publiccloud/basetest.pm:113]: looks like some debug code which you forgot to clean up

Comment (asmorodskyi) [reply lib/publiccloud/basetest.pm:113]: and two other record_info below looks the same

Comment (grisu48) [reply lib/publiccloud/basetest.pm:113]: Yeah, I marked the same but then realized that it's marked as notready therefore Pavel is probably still fiddling around here :slightly_smiling_face:

File: tests/console/gpg.pm [hunk 2]
```diff
@@ -192,7 +192,7 @@ sub run {
 }
 
 sub test_flags {
-    return {fatal => 0};
+    return {fatal => 1};
```
Comment (jlausuch) [inline tests/console/gpg.pm:195]: how is this related to this PR?

**Review by jlausuch — APPROVED** [review]
At first I was worried of so many changes, but it's a lot of small changes in imports and other things. Good work!

File: tests/console/gpg.pm (hunk 2 above)
Comment (pdostal) [reply tests/console/gpg.pm]: It's not. Thank you Jose

File: lib/main_publiccloud.pm [hunk 3]
```diff
@@ -60,20 +60,21 @@ sub load_maintenance_publiccloud_tests {
...
-    loadtest 'console/sudo';
-    loadtest 'console/supportutils';
-    loadtest 'console/journalctl';
-    loadtest 'console/procps';
-    loadtest 'console/suse_module_tools';
-    loadtest 'console/libgcrypt' unless is_sle '=12-SP4';
+    my ($run_args) = @_;
+    loadtest 'console/cleanup_qam_testrepos', run_args => $run_args unless get_var('PUBLIC_CLOUD_QAM');
+    loadtest 'console/openvswitch', run_args => $run_args;
+    loadtest 'console/rpm', run_args => $run_args;
```
Comment (asmorodskyi) [inline lib/main_publiccloud.pm:66]: I just randomly checked `console/rpm` it does not accept args . https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/rpm.pm#L31 . I am against passing it just to be generic ...

**Review by asmorodskyi — CHANGES_REQUESTED** [review]
let's clarify why we passing args for consoletests

File: lib/main_publiccloud.pm (hunk 3 above)
Comment (asmorodskyi) [reply lib/main_publiccloud.pm:66]: ah so you need it in `opensusebasetest->post_fail_hook` ?

Comment (pdostal) [reply lib/main_publiccloud.pm:66]: > ah so you need it in `opensusebasetest->post_fail_hook` ?

Yes. See [poo#95780](https://progress.opensuse.org/issues/95780).

Comment (asmorodskyi) [reply lib/main_publiccloud.pm:66]: ok

**Review by asmorodskyi — APPROVED** [review]
LGTM

File: tests/console/openssl_alpn.pm [hunk 4]
```diff
@@ -40,4 +40,10 @@ sub run {
     save_screenshot;
 }
 
+#sub test_flags {
```
Comment (asmorodskyi) [inline tests/console/openssl_alpn.pm:43]: please remove this

File: tests/console/rpm.pm [hunk 5]
```diff
@@ -36,6 +36,7 @@ sub run {
     # Download dummy test packages
     # wget is not present in opensuse-15.1
     assert_script_run("curl -o ${dir_prefix}openqa_rpm_test-1.0-0.noarch.rpm " . autoinst_url . "/data/rpm/openqa_rpm_test-1.0-0.noarch.rpm");
+    #assert_script_run("false");
```
Comment (asmorodskyi) [inline tests/console/rpm.pm:39]: please remove this

File: tests/console/gpg.pm [hunk 6]
```diff
@@ -191,8 +191,4 @@ sub run {
     }
 }
 
-sub test_flags {
```
Comment (asmorodskyi) [inline tests/console/gpg.pm:194]: I think this change influence not only execution in Public Cloud but also "normal" run of this test . Are we sure about consequences of this change ? ( I assume failure in gpg test will halt further  execution )

Comment (asmorodskyi) [reply tests/console/gpg.pm:194]: maybe we also need here `get_var('PUBLIC_CLOUD') ?` to limit change only to pub cloud ?

Comment (pdostal) [reply tests/console/gpg.pm:194]: Actually not, because `lib/consoletests.pm` is based on `lib/opensusebasetest.pm` where is the same code already present.
